src/blog/_posts/2018-09-26-opera-56.md

```diff
@@ -0,0 +1,155 @@
+colour depends on the angular distance to a line through the new
+[`conic-gradient`](https://www.w3.org/TR/css-images-4/#conic-gradients)
+property.
+
+Example:
+
+<div style="border: 3px solid black; background: conic-gradient(red, yellow, green, red); width: 400px; height: 400px;"></div>
+
+### Logical margin, paddings and border properties
+
+Blink, the Chromium/Opera browser engine, has gotten some [new CSS
+properties](https://drafts.csswg.org/css-logical-1/#box) that adapt to
+text direction. So `padding-inline-start` will apply to either left or
+right depending on what side you start reading the start. Other
+examples of newly added properties are `border-block-end-color`,
+`margin-inline-start` and `padding-inline-end`.
+
+### CSS Scroll Snap
+
+[CSS Scroll Snap](https://drafts.csswg.org/css-scroll-snap) allows the
+page author to determine what is suitable places in a document to snap
+to. This could be used to avoid cutting off content or to align
+content that is as large as the viewport correctly. This is a common
+problem in a tile based user interface.
+
+Previously this problem has been solved through JavaScript but doing
+it in CSS allows a much smoother user experience.
+
+### Display cutouts
+
+"Cutouts" is an area on a mobile phone where only part of the screen
+is available to the web page, like the notch at the top of some
+phones. With the display cutout support, it becomes possible to layout
+content around the cutout in an adaptive way.
+
+In CSS the necessary data can be extracted from the `env`
+function. See [the specification](https://drafts.csswg.org/css-env-1/)
+for details.
+
+### CSS Grid Layout
+
+The interpretation of height percentages for row tracks and gutters
+will change to be compliant with the specification in the next
+release, but already in Opera 56 you will see warnings if you are
+using height calculations that might change.
+
+## Media
+
+### OffscreenCanvas
+
+OffscreenCanvas is a canvas intended to be used in workers. Sometimes
+blitting requires heavy calculation more suitable for a webworker and
+then OffsceenCanvases will come in handy. They have almost the [same
+API](https://html.spec.whatwg.org/#the-offscreencanvas-interface) as
+ordinary canvases so moving code to a worker should be easy.
+
+### `stalled` removed from HTMLMediaElements
+
+There used to be an event `stalled` that fired if no data had been
+received for 3 seconds. This was not useful since the client can be
+fine receiving data less often depending on buffer sizes and transfer
+chunk sizes. The event has now been removed.
+
+### EME (Encrypted Media Extensions)
+
+It is now possible for a web developer to query the browser about what
+encryption schemes it supports through EME.
+
+## DOM
+
+### `Element.toggleAttribute()` added
+
+With
+[`Element.toggleAttribute`](https://dom.spec.whatwg.org/#dom-element-toggleattribute)
+an attribute can be removed if it doesn't exist or added if it does
```

Comment:
"an attribute can be removed if it doesn't exist or added if it does"

...the other way around?